### PR TITLE
Add iterable classes and `values()` implementation to `FakeQuerySet`

### PR DESCRIPTION
**Pull request by ababic**
This PR adds the following:
- Django-like 'iterable' classes that return values in the expected format - allowing us to keep the actual 'evaluation' (not really evaluation, but you know what I mean!) 'lazy' 
- A values() implementation

### CODE REVIEW

File: tests/tests/test_cluster.py [hunk 1]
```diff
@@ -478,6 +553,147 @@ def test_date_filters(self):
...
+            tuple(band.members.filter(favourite_restaurant__proprietor__name__iexact="gORDON rAMSAY")),
+            (band.members.get(name="John Lennon"),)
+        )
+        self.assertEqual(
+            tuple(band.members.filter(favourite_restaurant__proprietor__name__iexact="mARCO pIERRE wHITE")),
+            (band.members.get(name="Ringo Starr"),)
+        )
+        # Using an exact proprietor comparisson
+        self.assertEqual(
+            tuple(band.members.filter(favourite_restaurant__proprietor=gordon)),
```
Comment (zerolab) [inline tests/tests/test_cluster.py:608]:
```suggestion
            tuple(band.members.filter(favourite_restaurant__proprietor=self.gordon_ramsay)),
```

File: tests/tests/test_cluster.py [hunk 2]
```diff
@@ -265,10 +343,7 @@ def test_queryset_filtering(self):
         self.assertEqual(3, beatles.members.filter(band=also_beatles).count())
 
     def test_queryset_filtering_on_models_with_inheritance(self):
-        strawberry_fields = Restaurant.objects.create(name='Strawberry Fields')
-        the_yellow_submarine = SeafoodRestaurant.objects.create(name='The Yellow Submarine')
-
-        john = BandMember(name='John Lennon', favourite_restaurant=strawberry_fields)
+        john = BandMember(name='John Lennon', favourite_restaurant=self.strawberry_fields)
```
Comment (zerolab) [inline tests/tests/test_cluster.py:346]:
```python
        # queried instance is more specific
        self.assertEqual(
            list(beatles.members.filter(favourite_restaurant=the_yellow_submarine)),
            [ringo]
        )
```
several lines below needs to be 


```python
        # queried instance is more specific
        self.assertEqual(
            list(beatles.members.filter(favourite_restaurant=self.the_yellow_submarine)),
            [ringo]
        )
```

**Comment by ababic** [comment]
Thanks @zerolab - all sorted now :)

**Comment by ababic** [comment]
I promise I'm 100% done with this now :)

**Review by gasman — APPROVED** [review]
Nicely done! I really like the approach to switching out the iterable classes while keeping the rest of FakeQuerySet intact, and the way that the `extract_field_value` helper from the previous PR so neatly drops into this.

**Comment by ababic** [comment]
Thank you @gasman! Glad to be of help :)

**Pull request merged**